packages/core-data/src/hooks/use-query-select.ts

```diff
@@ -71,10 +72,28 @@ interface QuerySelectResponse {
  * @return {QuerySelectResponse} Queried data.
  */
 export default function __experimentalUseQuerySelect( mapQuerySelect, deps ) {
-	return useSelect( ( select, registry ) => {
-		const resolve = ( store ) => enrichSelectors( select( store ) );
-		return mapQuerySelect( resolve, registry );
-	}, deps );
+	const { data, isResolving, hasResolved, ...rest } = useSelect(
+		( select, registry ) => {
+			const resolve = ( store ) => enrichSelectors( select( store ) );
+			return mapQuerySelect( resolve, registry );
+		},
+		deps
+	);
+
+	let status;
+	if ( isResolving ) {
+		status = Status.Resolving;
+	} else if ( hasResolved ) {
+		if ( data ) {
```

Comment:
That's far from ideal, but it's the best we can do until there's a better [resolution failure handling](https://github.com/WordPress/gutenberg/pull/38669) in place.